app/DynamicsAdapter/DynamicsAdapter.Web/SearchAgency/AgencyRequestController.cs

```diff
@@ -81,6 +82,20 @@ public async Task<IActionResult> CreateSearchRequest(string requestId, [FromBody
                     _logger.LogError(ex.Message);
                     return StatusCode(StatusCodes.Status500InternalServerError, new { ReasonCode = ex.Message, Message = ex.InnerException?.Message });
                 }
+
+                //try to submit to queue and then get EstimatedDate and positionInQueue
+                try
+                {
+                    await _agencyRequestService.SubmitSearchRequestToQueue(createdSearchRequest.SearchRequestId);
+                    createdSearchRequest = await _agencyRequestService.RefreshSearchRequest(createdSearchRequest.SearchRequestId);
+                }catch(Exception e)
+                {
+                    _logger.LogError(e, "submit to queue or get current search request failed.");
```

Comment:
There are one of two things here that cold have failed. the error message is basically saying I did a bunch of stuff and something failed.   Seeing that error message, what is the support person supposed to do?  You could track which of the two operations worked and didn't and provide a more specific error message.